File: tests/conformance/fixtures/go/03-semantic.go
package main

import sykli "github.com/yairfalse/sykli/sdk/go"

func main() {
	p := sykli.New()

	p.Task("test-auth").Run("pytest tests/auth/").
		Inputs("src/auth/**/*.py", "tests/auth/**/*.py").
		Covers("src/auth/*").
		Intent("unit tests for auth module").
		SetCriticality("high").
		OnFail("analyze").
		SelectMode("smart")

	p.Emit()
}

File: tests/conformance/fixtures/typescript/03-semantic.ts
import { Pipeline } from '../../../../sdk/typescript/src/index';

const p = new Pipeline();

p.task('test-auth').run('pytest tests/auth/')
  .inputs('src/auth/**/*.py', 'tests/auth/**/*.py')
  .covers('src/auth/*')
  .intent('unit tests for auth module')
  .setCriticality('high')
  .onFail('analyze')
  .selectMode('smart');

p.emit();
